File: dentis-frontend/src/app/core/services/appointment.service.ts
import { Injectable } from '@angular/core';
import { HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiService, PageResponse } from './api.service';
import { Appointment, AppointmentStatus, CreateAppointmentRequest } from '../models/appointment.model';

@Injectable({ providedIn: 'root' })
export class AppointmentService {
  constructor(private api: ApiService) {}

  create(request: CreateAppointmentRequest): Observable<Appointment> {
    return this.api.post<Appointment>('/appointments', request);
  }

  findById(id: string): Observable<Appointment> {
    return this.api.get<Appointment>(`/appointments/${id}`);
  }

  findByPatient(patientId: string, page = 0, size = 20): Observable<PageResponse<Appointment>> {
    const params = new HttpParams().set('page', page).set('size', size);
    return this.api.get<PageResponse<Appointment>>(`/appointments/patient/${patientId}`, params);
  }

  getDentistCalendar(dentistId: string, from: string, to: string): Observable<Appointment[]> {
    const params = new HttpParams().set('from', from).set('to', to);
    return this.api.get<Appointment[]>(`/appointments/dentist/${dentistId}/calendar`, params);
  }

  updateStatus(id: string, status: AppointmentStatus): Observable<Appointment> {
    const params = new HttpParams().set('status', status);
    return this.api.patch<Appointment>(`/appointments/${id}/status?status=${status}`);
  }

  cancel(id: string): Observable<void> {
    return this.api.delete<void>(`/appointments/${id}`);
  }
}

File: dentis-application/src/main/java/com/dentis/application/service/AppointmentService.java
package com.dentis.application.service;

import com.dentis.application.dto.request.CreateAppointmentRequest;
import com.dentis.application.dto.request.UpdateAppointmentRequest;
import com.dentis.application.dto.response.AppointmentResponse;
import com.dentis.common.response.PageResponse;
import com.dentis.domain.enums.AppointmentStatus;
import org.springframework.data.domain.Pageable;

import java.time.LocalDateTime;
import java.util.List;

public interface AppointmentService {

    AppointmentResponse create(CreateAppointmentRequest request);

    AppointmentResponse update(String id, UpdateAppointmentRequest request);

    AppointmentResponse findById(String id);

    PageResponse<AppointmentResponse> findByPatient(String patientId, Pageable pageable);

    PageResponse<AppointmentResponse> findByDentist(String dentistId, Pageable pageable);

    List<AppointmentResponse> findDentistCalendar(String dentistId, LocalDateTime from, LocalDateTime to);

    AppointmentResponse updateStatus(String id, AppointmentStatus status);

    void cancel(String id);
}
